packages/engine/paima-sm/src/index.ts

```diff
@@ -481,15 +554,17 @@ async function processScheduledData(
           }
         });
       }
+      resultingHashes.successTxHashes.push(txHash);
```

Comment:
this is wrong (here and below). We have to instead look at the return value of `tryOrRollback`